File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        
        for(int i=0;i<citations.length;i++){
            int h = Math.min(citations[i],citations.length-i);
            if(citations[i]>=answer)
                answer= Math.max(answer,h);
            else
                break;
        
        }
        return answer;
    }
}

File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    for h in range(1,len(citations)+1):
        cnt = 0
        for x in citations:
        # h번 이상 인용된 논문이 h편 이상이라면
            if h <= x:
                cnt +=1
        if h<=cnt:
            answer = h
            
    return answer